### ISSUE TITLE
`generate-config` produces invalid configuration - DraftDataset with blank org

### ISSUE THREAD

**Issue by rustyjux** [bcgov/gwa-cli#127]
## Description
When using `gwa generate-config` interactively, if org/orgunit are left blank, then they are blank in the created DraftDataset and validation fails when publishing.

```
organization: 
organizationUnit:
```

Meanwhile, if parameters are provided (e.g.
`gwa generate-config --template quick-start --service <MYSERVICE> --upstream https://httpbin.org`) then org/orgunit are autofilled with:
```
organization: ministry-of-citizens-services
organizationUnit: databc
```
 
org and orgunit are optional fields (ie you can omit those fields and still add/update a dataset), but they cannot be blank.

## Acceptance criteria
- change the behaviour for `generate-config` to never create DraftDataset where `organization` and `organizationUnit` are blank which will fail validation
  - do not ask for org/orgunit in the interactive mode of generate-config (new behaviour)
  - have org and orgunit default to CITZ/databc if creating a dataset from both interactive (new behaviour) and parameters provided (current behaviour)

## Other thoughts and considerations
We agree interactive and provided parameters **should behave the same** and should not leave blank fields in yaml.

Other approaches considered to resolve this issue:
  - A) provide default values (CITZ/databc) to interactive, hit ENTER to accept
  - B) do not have default values, publish dataset without org/orgunit if not provided
  - C) don't prompt for org/orgunit when using `generate-config`
  - D) modify API to handle blanks values

**Comment by ikethecoder** [comment]
I think making the two approaches (interactive and parameters) consistent is necessary and in the spirit of making valid output, defaulting org and org-unit.  Not sure if it needs to be added to the interactive prompts, unless every Dataset field is part of the interactive experience already.

I think longer term datasets will be created under an Organization, so I don't imagine a situation where a dataset would have no Org/OrgUnit, so don't think allowing blank Org/Orgunit is acceptable.

**Comment by rustyjux** [comment]
I agree on the consistency - hence the bold I added on that point. The only dataset fields in the interactive prompts are org and orgunit and they (humorously?) show `(optional)` as a hint:
![image](https://github.com/user-attachments/assets/45d05d0d-0811-4cef-a144-ff5b512b77ae)
![image](https://github.com/user-attachments/assets/07d64942-8367-493b-b74f-26a6c5f60bc4)

I like the idea of removing them from `generate-config` and going with defaults. But this begs the question of how and when do folks update the organization on the dataset? Which leads me to ask - shouldn't the dataset org always align with the Gateway org and why are we making folks pick an org more than once? (rather than using the Gateway org when creating a dataset). 

Assuming dataset org should equal Gateway org, this begs the question of when do teams add their Gateway to an org? (current answer: when they want to publish to the API Directory). **Should picking an org be part of creating a Gateway initially?**

Possible but unlikely edge case: could there be a team with a Gateway that supports a few orgunits?

**Comment by ikethecoder** [comment]
re: "team with a Gateway that supports a few orgunits" - that has happened before and the solution was to create separate Gateways.  Gateway can only be owned by one Ministry.

re: Should picking an org be part of creating a Gateway initially? - along same lines as org/orgunit, we want "getting started" to be as frictionless as possible, so having them choose one is another bit of data they have to think about.

re: dataset org always align with the Gateway org - this is true and enforced (can't publish to the API Directory if the org/orgunit are mismatched).  But I think what we perhaps have to flush out is what is our desired state for the user experience - what is the order that things happen - my thought is that Ministries can have a catalog of Datasets (perhaps wrong terminology?) and when a Gateway is assigned to the Ministry, the Gateway API Provider is then able to link API Products to the Datasets that are part of that Ministry.

**Comment by rustyjux** [comment]
Good point we don't want picking org to be part of creating a Gateway because of friction on entry.

With regards to this ticket, maybe we can agree that (while not entirely ideal) we will:
- not ask for org/orgunit in the interactive mode of `generate-config` (new)
- have org and orgunit default to CITZ/databc in both interactive (new) and parameters provided (current)

The consequence is most API providers (not being part of DSS) will need to update the dataset org and orgunit. They can do this when they go to publish and will be blocked from enabling at this time.
  - error message in Portal - good!
    ![image](https://github.com/user-attachments/assets/c7cb3ce8-5371-4561-8899-7a622aab075e)
  - error message in CLI - not great (debug doesn't help)
    ![image](https://github.com/user-attachments/assets/43433e97-3463-42df-a259-a2e5fa656188)

Re: enforcement of Gateway org/Dataset org - not a big deal but there is a workaround to skip this. I wanted to see where this is validated and found if you have an enabled/active dataset (in your org) you can change the org on it. Validation must happen only when the Environment is first enabled.

So possible follow up is to further improve error messaging in CLI. And fix that workaround if we care but I don't think it's worthy.

**Comment by ikethecoder** [comment]
sounds good!

**Comment by rustyjux** [comment]
This ticket is ready 🤘

**Comment by rustyjux** [comment]
test GitHub-Jira gha

**Comment by rustyjux** [comment]
Released in v3.0.6
